packages/core/src/utils/fields.ts

```diff
@@ -1,22 +1,73 @@
-import { CardSchema, CompiledCard, Field } from '../interfaces';
 import { isNotReadyError } from './errors';
 import set from 'lodash/set';
 import get from 'lodash/get';
 import isPlainObject from 'lodash/isPlainObject';
+import { FieldsWithPlaceholders, FieldWithPlaceholder } from '../compiler';
+import { CardSchema, ComponentInfo, Format } from '../interfaces';
+import { TemplateUsageMeta } from '../glimmer-plugin-card-template';
 
-export function getFieldForPath(fields: CompiledCard['fields'], path: string): Field | undefined {
+export function getFieldForPath(fields: FieldsWithPlaceholders, path: string): FieldWithPlaceholder | undefined {
   let paths = path.split('.');
   let [first, ...tail] = paths;
 
   let field = fields[first];
 
-  if (paths.length > 1) {
+  if (paths.length > 1 && field.card !== 'self') {
     return getFieldForPath(field.card.fields, tail.join('.'));
   }
 
   return field;
 }
 
+export function buildUsedFieldsListFromUsageMeta(
```

Comment:
it looks like this moved but an upstream merge moved this back here. I'll fix that